packages/amplify-migration-codegen-e2e/src/helpers.ts

```diff
@@ -0,0 +1,163 @@
+  deleteProjectDir,
+  initJSProjectWithProfile,
+  addAuthWithDefault,
+  amplifyPushAuth,
+  getProjectMeta,
+  getUserPool,
+} from '@aws-amplify/amplify-e2e-core';
+import { npmInstall, getNpxPath, nspawn as spawn } from '@aws-amplify/amplify-e2e-core';
+import * as fs from 'fs-extra';
+import { $TSAny } from '@aws-amplify/amplify-cli-core';
+import path from 'node:path';
+import assert from 'node:assert';
+import { CloudControlClient, GetResourceCommand } from '@aws-sdk/client-cloudcontrol';
+
+const pushTimeoutMS = 1000 * 60 * 20; // 20 minutes;
+
+export async function getResourceDetails(typeName: string, identifier: string, region: string) {
+  const client = new CloudControlClient({ region });
+  const command = new GetResourceCommand({
+    TypeName: typeName,
+    Identifier: identifier,
+  });
+
+  try {
+    const response = await client.send(command);
+    return JSON.parse(response.ResourceDescription.Properties);
+  } catch (error) {
+    console.error('Error fetching resource details:', error);
+    throw error;
+  }
+}
+
+export function deployGen2CDK(cwd: string) {
+  try {
+    npmInstall(cwd);
+  } catch (error) {
+    console.error('Failed to install dependencies:', error);
+    throw error;
+  }
+  return spawn(getNpxPath(), ['ampx', 'sandbox', '--once'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  }).runAsync();
+}
+
+export function migrateCommand(cwd: string) {
+  return spawn(getNpxPath(), ['@aws-amplify/migrate', 'to-gen-2', 'generate-code'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  }).runAsync();
+}
+
+export function deleteGen2Project(cwd: string) {
+  return spawn(getNpxPath(), ['ampx', 'sandbox', 'delete'], {
+    cwd,
+    stripColors: true,
+    noOutputTimeout: pushTimeoutMS,
+    env: { ...process.env, npm_config_user_agent: 'npm' },
+  })
+    .wait("Are you sure you want to delete all the resources in your sandbox environment (This can't be undone)?")
+    .sendConfirmYes()
+    .wait('Finished deleting.')
+    .runAsync();
+}
+
+export async function setupGen1Project(projRoot: string, projectName: string) {
+  await initJSProjectWithProfile(projRoot, { name: projectName, disableAmplifyAppCreation: false });
+  await addAuthWithDefault(projRoot);
+  await amplifyPushAuth(projRoot);
+}
+
+export async function assertGen1Setup(projRoot: string) {
+  const gen1Meta = getProjectMeta(projRoot);
+  const gen1UserPoolId = Object.keys(gen1Meta.auth).map((key) => gen1Meta.auth[key])[0].output.UserPoolId;
+  const gen1Region = gen1Meta.providers.awscloudformation.Region;
+  const userPool = await getUserPool(gen1UserPoolId, gen1Region);
+  expect(userPool.UserPool).toBeDefined();
+  return { gen1UserPoolId, gen1Region };
+}
+
+export async function assertGen2Resources(projRoot: string, gen1UserPoolId: string, gen1Region: string) {
```

Comment:
make it more specific `assertGen2UserPoolResource` since later we will have other ones.